### PR TITLE
channel details, channel delete modal

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

https://app.clickup.com/t/6a5vaw
https://app.clickup.com/t/6a5vav

**PR intended to be tested with API branch:** feature/multichannel-multicurrency

### Screenshots
Screenshot 2020-07-09 at 16 53 00
Screenshot 2020-07-09 at 16 55 56

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CU-6a5vaw User needs to be able to edit channels](https://app.clickup.com/t/6a5vaw)

**Comment by patrys** [comment]
Task linked: [CU-6a5vav User needs to be able to delete channels](https://app.clickup.com/t/6a5vav)

File: src/channels/views/ChannelsList/ChannelsList.tsx [hunk 1]
```diff
@@ -14,23 +18,54 @@ interface ChannelsListProps {
...
+  const channelsChoices = useMemo(
+    () =>
+      params.id &&
+      data?.channels
+        ?.map(channel => ({
+          label: channel.name,
+          value: channel.id
+        }))
+        .filter(channel => channel.value !== params.id),
+    [data?.channels?.length, params.id]
```
Comment (dominik-zeglen) [inline src/channels/views/ChannelsList/ChannelsList.tsx:34]: I'm not sure if depending on arrays length is good enough - let's say that one channel got updated, but length is still the same. Choices are not updated.
TBH I don't see any reason for this to be memoized, creating this array is, well, free.

File: src/channels/views/ChannelDetails/ChannelDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        })}
+      />
+      <Container>
+        <AppHeader onBack={handleBack}>
+          {intl.formatMessage(sectionNames.channels)}
+        </AppHeader>
+        <PageHeader
+          title={
+            <FormattedMessage
+              defaultMessage="{name}"
```
Comment (dominik-zeglen) [inline src/channels/views/ChannelDetails/ChannelDetails.tsx:72]: I don't think this needs to be translated :)

File: src/channels/views/ChannelsList/ChannelsList.tsx [hunk 3]
```diff
@@ -14,23 +18,52 @@ interface ChannelsListProps {
...
+      ?.map(channel => ({
+        label: channel.name,
+        value: channel.id
+      }))
+      .filter(channel => channel.value !== params.id);
+
   const navigateToChannelCreate = () => navigate(channelAddUrl);
 
-  const onRemove = () => null;
+  const handleRemoveConfirm = () => null;
```
Comment (orzechdev) [inline src/channels/views/ChannelsList/ChannelsList.tsx:36]: It won't remove the channel, but the referenced task explains it should -  [ https://app.clickup.com/t/6a5vav](https://app.clickup.com/t/6a5vav)

Comment (AlicjaSzu) [reply src/channels/views/ChannelsList/ChannelsList.tsx:36]: I updated task description

**Pull request merged**